packages/actor-http-cache/lib/ActorHttpCache.ts

```diff
@@ -0,0 +1,233 @@
+    const response = await this.fetchDocument({
+      ...action,
+      init: {
+        ...action.init,
+        headers: revalidationHeaders,
+      },
+    });
+    const { policy, modified } = this.getRevalidatedPolicy(newRequest, response, cacheResult.policy);
+    const finalResponse = modified ? response : new Response(cacheResult.body, cacheResult.init);
+    return await this.teeAndCacheResponse(newRequest, finalResponse, policy, cacheStorage);
+  }
+
+  private async fetchDocument(action: IActionHttp): Promise<Response> {
+    const response = await this.mediatorHttp.mediate({
+      ...action,
+      context: action.context.set(KeysHttpCache.doNotCheckHttpCache, true),
+    });
+    return response;
+  }
+
+  private newCachePolicy(request: Request, response: Response): CachePolicy {
+    // Conversions to make headers compatible
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    const responseWithHashHeaders = ActorHttpCache.responseToResponseWithHashHeaders(response);
+    return new CachePolicy(
+      requestWithHashHeaders,
+      responseWithHashHeaders,
+    );
+  }
+
+  private isPolicySatisfied(request: Request, policy: CachePolicy): boolean {
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    return policy.satisfiesWithoutRevalidation(requestWithHashHeaders);
+  }
+
+  private getRevalidationHeaders(request: Request, policy: CachePolicy): Headers {
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    const hashRevalidationHeaders = policy.revalidationHeaders(requestWithHashHeaders);
+    return new Headers(<Record<string, string>> hashRevalidationHeaders);
+  }
+
+  private getRevalidatedPolicy(request: Request, response: Response, policy: CachePolicy): RevalidationPolicy {
+    const requestWithHashHeaders = ActorHttpCache.requestToRequestWithHashHeaders(request);
+    const responseWithHashHeaders = ActorHttpCache.responseToResponseWithHashHeaders(response);
+    return policy.revalidatedPolicy(requestWithHashHeaders, responseWithHashHeaders);
+  }
+
+  private async teeAndCacheResponse(
+    request: Request,
+    response: Response,
+    cachePolicy: CachePolicy,
+    cacheStorage: IHttpCacheStorage,
+  ): Promise<Response> {
+    // Node-fetch does not support body.tee, while it is mandatory according to the fetch and readablestream api.
+    // If it doesn't exist, we monkey-patch it.
+    if (response.body && !response.body.tee) {
+      response.body.tee = (): [ReadableStream, ReadableStream] => {
+        // eslint-disable-next-line import/no-nodejs-modules
+        const Readable = require('stream').Readable;
+        return [
+          Readable.from(response.body),
+          Readable.from(response.body),
+        ];
+      };
+    }
+
+    // There almost will always be a body, so ignore next
```

Comment:
I'm not a big fan of ignoring coverage :P
Shouldn't be too much effort to test this case as well?